rosbag2/src/rosbag2/sequential_reader.cpp

```diff
@@ -71,9 +76,34 @@ SequentialReader::open(
   }
 }
 
+std::string SequentialReader::get_next_file()
+{
+  {
+    assert(current_file_offset_ < file_paths_.size());
+    current_file_offset_++;
+    return file_paths_.at(current_file_offset_);
+  }
```

Comment:
iterator could make things easier here as `get_next_file()` essentially would only be dereferencing it